encode.go

```diff
@@ -352,7 +355,7 @@ var bytesType = reflect.TypeOf([]byte{})
 
 func (en *encoder) sliceReflect(key uint64, slval reflect.Value) {
 
-	if slval.Kind() != reflect.Slice {
+	if slval.Kind() != reflect.Slice && slval.Kind() != reflect.Array {
```

Comment:
maybe insert a "kind := slval.Kind()" before and test that variable, to avoid calling possibly expensive reflection method Kind() twice in a row unnecessarily.